### DISCUSSION TITLE
A lot of ideas

### DISCUSSION THREAD

**Discussion by Miala-python** [ScratchAddons/ScratchAddons#3680] [Ideas]
Hello,
I have a lot of ideas for ScratchAddons and before make issues I propose all.

> _Sorry if my English isn't perfect, I'm French XD _

/!\ upload = Computer -> Scratch
/!\ download = Scratch -> Computer

My list of ideas:
- Keyboard shortcut (mostly for the pause button [Alt+P])
- Loading bar for upload thumbnail and sprites from computer
- Upload the cloud variables when you upload sprite
- Modify the variable's type
- Copy and paste to different tabs for costumes editor
- Make animations for Halloween and Noel. (like dango)
- Add hacked blocks (view the folder HB in https://scratch.mit.edu/projects/563266908/editor/ )
- Change users for when you have multi-accounts
- Make an icon for save (like LibreOffice), it will take less space
- Add **'Turn on Slow Mode'** and **'Turn on ☁ test Mode' (4 stages in the same window/tab)** in Edit menu
- And make a button [more ▾] with the buttons:
> - Tutorials
> - Put a thumbnail
> - Share / Unshare

And this new buttons:

> - Change all the blocks **When ⚑ is clicked** as **When I receive [⚑ ▾]**
> - Translate variables's, blocks's and messages's name in [French, English, Spanish, ... ▾]
> - HTMLifier (view https://sheeptester.github.io/htmlifier/ )

Thank's !

**Comment by cobaltt7** [comment]
Ø  Upload the cloud variables when you upload sprite

  *   Modify the variable's type

Ø  Add 'Turn on Slow Mode' and 'Turn on ☁ test Mode' (4 stages in the same window/tab) in Edit menu

  *   Translate variables's, blocks's and messages's name in [French, English, Spanish, ... ▾]

Can you clarify these?

  *   Make animations for Halloween and Noel. (like dango)

Probably not going to happen, Dango rain was for April Fools, the prank holiday.

Maybe we could add an option to customize the emoji though 🤔

  *   Add hacked blocks (view the folder HB in https://scratch.mit.edu/projects/563266908/editor/ )

#817

  *   Change users for when you have multi-accounts

#3327

  *   Make an icon for save (like LibreOffice), it will take less space

Ø  And make a button [more ▾] with the buttons:

#2110

  *   Change all the blocks When ⚑ is clicked as When I receive [⚑ ▾]

Not useful

Ø  HTMLifier (view https://sheeptester.github.io/htmlifier/ )

#762

**Reply by GrahamSH-LLK** [reply]
*   Change all the blocks When ⚑ is clicked as When I receive [⚑ ▾]
this is actually quite useful, but I'm not sure if it's important enough for it's own addon

**Reply by Dhuls-Scratch** [reply]
IMO, [Turbowarp Packager](https://packager.turbowarp.org) is better, and the compiler can be disabled if you want it more like Scratch.

**Reply by Miala-python** [reply]
> IMO, [Turbowarp Packager](https://packager.turbowarp.org) is better, and the compiler can be disabled if you want it more like Scratch.

Okay, thank you

**Reply by Miala-python** [reply]
What means * and Ø ?

**Comment by sfederici** [comment]
>     * Keyboard shortcut (mostly for the pause button [Alt+P])

Full support. For "keyboard" people like me, having to move each time to the pause button (that I use a lot when evaluating my students' projects) is really annoying. Having a keyboard shortcut to start/stop the pause would save me a lot of time.

**Reply by Samq64** [reply]
See #5121

**Comment by sfederici** [comment]
> [rephrased] Translate name of variables, custom blocks and messages to another language

This -that could be somewhat useful for a very well-known language, worldwide, such as English- is useful for less common languages like, e.g., french or german, but it is **extremely** useful for very "distant" languages, such as the eastern languages like chinese, japanese, korean, etc. Understanding the name (and then the meaning) of variables, custom blocks and messages would be an incredible help to learn (remix) from very good projects that, otherwise, will remain inaccessible.
